#For NTU VIRAL dataset, using sensors: 

ilive_common:
   if_dump_log: 0                   # If recording ESIKF update log. [default = 0]
   record_offline_map: 1            # If recording offline map. [default = 1]
   pub_pt_minimum_views: 3          # Publish points which have been render up to "pub_pt_minimum_views" time. [default = 3]
   minimum_pts_size: 0.01           # The minimum distance for every two points in Global map (unit in meter). [default = 0.01] 
   image_downsample_ratio: 1        # The downsample ratio of the input image. [default = 1]
   estimate_i2c_extrinsic: 1        # If enable estimate the extrinsic between camera and IMU. [default = 1]
   estimate_intrinsic: 1            # If enable estimate the online intrinsic calibration of the camera lens. [default = 1] 
   maximum_vio_tracked_pts: 600     # The maximum points for tracking. [default = 600]
   append_global_map_point_step: 4  # The point step of append point to global map. [default = 4]
   gravity: [ 0 , 0 , 9.805 ]       # Gravity
   esikf_iter_f2f_times: 2       #pnp
   esikf_iter_f2map_times: 2     #photometric
   use_lio: 1
   use_vio_f2f: 1
   use_vio_f2map: 1
   if_save_mapping: false
   method: "EKF"                         #EKF,InEKF,EIKF



ilive_vio:
   image_width: 1280
   image_height: 1024
   camera_intrinsic:
      [863.4241, 0.0, 640.6808,
      0.0,  863.4171, 518.3392,
      0.0, 0.0, 1.0 ] 
   camera_dist_coeffs: [-0.1080, 0.1050, -1.2872e-04, 5.7923e-05, -0.0222]  #k1, k2, p1, p2, k3
   # Fine extrinsic value. form camera-LiDAR calibration.
   camera_ext_R:
         [-0.00113207, -0.0158688, 0.999873,
            -0.9999999,  -0.000486594, -0.00113994,
            0.000504622,  -0.999874,  -0.0158682]
   camera_ext_t: [0.050166, 0.0474116, -0.0312415] 
   # camera_ext_t: [0,0,0] 

   cam_cov_scale: 1
   camera_time_delay: 0   # t_imu = t_cam + t_shift


Lidar_front_end:
   lidar_type: 1   # 0 for mid, 1- Livox-horizon, 2 for velodyne 16, ï¼Œ3 for Ouster
   N_SCANS: 6
   using_raw_point: 1
   point_step: 1
   lid_cov_scale: 0.00015


ilive_lio:        
   lio_update_point_step: 4            # Point step used for LIO update.
   max_iteration: 3          # Maximum times of LIO esikf. The number of the iterative steps.//1.2.3
   lidar_time_delay: 0        # The time-offset between LiDAR and IMU, provided by user. 
   voxel_downsample_size_surf: 0.3
   voxel_downsample_size_axis_z: 0.3
   filter_size_corner: 0.30   
   filter_size_surf: 0.30
   filter_size_surf_z: 0.30
   filter_size_map: 0.30

   ext_t_lid_in_imu: [0.04165,0.02326,-0.0284]
   ext_R_lid_in_imu:
              [1,0,0,
              0,1,0,
              0,0,1]
   publish_feature_map: false  
   long_rang_pt_dis: 500             
   planar_check_dis: 0.05                                                                           

ilive_IMU:
   max_IMU_calibration_num: 500
   imu_cov_scale: 1 #initial state scale setting
   # COV_ACC_NOISE_DIAG:      0.4
   # COV_GYRO_NOISE_DIAG:     0.2
   # COV_BIAS_ACC_NOISE_DIAG: 0.05
   # COV_BIAS_GYRO_NOISE_DIAG: 0.1
   COV_ACC_NOISE_DIAG:  1.3354056e-3
   COV_GYRO_NOISE_DIAG: 1.349803352e-3
   COV_BIAS_ACC_NOISE_DIAG: 1.87489e-5
   COV_BIAS_GYRO_NOISE_DIAG: 7.0756e-7